v4/weaviate/data/objectsGetter.go

```diff
@@ -48,29 +55,47 @@ func (getter *ObjectsGetter) WithLimit(limit int) *ObjectsGetter {
 
 // Do get the data object
 func (getter *ObjectsGetter) Do(ctx context.Context) ([]*models.Object, error) {
-	param := getAdditionalParams(getter.additionalProperties)
-	if getter.withLimit {
-		param += fmt.Sprintf("?limit=%v", getter.limit)
-	}
-	responseData, err := getObjectList(ctx, "/objects", getter.id, param, getter.connection)
+	responseData, err := getter.objectList(ctx)
 	if err != nil {
 		return nil, err
 	}
+
 	if responseData.StatusCode != 200 {
 		return nil, except.NewUnexpectedStatusCodeErrorFromRESTResponse(responseData)
 	}
+
 	if getter.id == "" {
 		var objects models.ObjectsListResponse
 		decodeErr := responseData.DecodeBodyIntoTarget(&objects)
 		return objects.Objects, decodeErr
 	}
+
 	var object models.Object
 	decodeErr := responseData.DecodeBodyIntoTarget(&object)
 	return []*models.Object{&object}, decodeErr
 }
 
-// getAdditionalParams build the query URL parameters for the requested underscore properties
-func getAdditionalParams(additionalProperties []string) string {
+func (getter *ObjectsGetter) objectList(ctx context.Context) (*connection.ResponseData, error) {
+	responseData, err := getter.connection.RunREST(ctx, getter.buildPath(), http.MethodGet, nil)
+	if err != nil {
+		return responseData, except.NewDerivedWeaviateClientError(err)
+	}
+	return responseData, err
+}
+
+func (getter *ObjectsGetter) buildPath() string {
+	basePath := buildObjectsPath(getter.id, getter.className)
+
+	params := buildAdditionalParams(getter.additionalProperties)
+	if getter.withLimit {
+		params += fmt.Sprintf("?limit=%v", getter.limit)
```

Comment:
if additional params exists, would the final params string be `?include=xxx?limit=yyy` instead `?include=xxx&limit=yyy`?

Reply:
good catch! looks like this implementation never worked properly if additional param was passed with limit